contracts/multisig-improved/src/action_types/external_module.rs

```diff
@@ -0,0 +1,80 @@
+use multiversx_sc_modules::transfer_role_proxy::PaymentsVec;
+
+multiversx_sc::imports!();
+
+pub type ModuleId = AddressId;
+
+mod external_module_proxy {
+    use multiversx_sc_modules::transfer_role_proxy::PaymentsVec;
+
+    multiversx_sc::imports!();
+
+    #[multiversx_sc::proxy]
+    pub trait ExternalModuleProxy {
+        #[view(canExecute)]
+        fn can_execute(
+            &self,
+            sc_address: ManagedAddress,
+            original_caller: ManagedAddress,
+            egld_value: BigUint,
+            esdt_payments: PaymentsVec<Self::Api>,
+        ) -> bool;
+    }
+}
+
+#[multiversx_sc::module]
+pub trait ExternalModuleModule:
+    crate::common_functions::CommonFunctionsModule + crate::state::StateModule
+{
+    fn can_execute_action(
+        &self,
+        sc_address: ManagedAddress,
+        egld_value: BigUint,
+        esdt_payments: PaymentsVec<Self::Api>,
+    ) -> bool {
+        let module_id_mapper = self.module_id();
+        let module_id = module_id_mapper.get_id(&sc_address);
+        if module_id == NULL_ID {
+            return false;
+        }
+
+        let original_caller = self.blockchain().get_caller();
+        for module_id in self.active_modules_ids().iter() {
+            let opt_module_address = module_id_mapper.get_address(module_id);
+            require!(opt_module_address.is_some(), "Invalid setup");
+
+            let module_address = unsafe { opt_module_address.unwrap_unchecked() };
+            let can_execute: bool = self
+                .external_sc_proxy(module_address)
+                .can_execute(
+                    sc_address.clone(),
+                    original_caller.clone(),
+                    egld_value.clone(),
+                    esdt_payments.clone(),
+                )
+                .execute_on_dest_context();
+
+            if can_execute {
+                return true;
+            }
+        }
+
+        false
+    }
+
+    #[proxy]
+    fn external_sc_proxy(
+        &self,
+        sc_address: ManagedAddress,
+    ) -> external_module_proxy::Proxy<Self::Api>;
+
+    #[storage_mapper("moduleId")]
```

Comment:
Just a q - why is the storage setup for a list of addresses this bloated?

Reply:
It's easier to work with IDs in the contract, i.e. `active_modules_ids` doesn't have to store addresses, it just stores IDs.

Reply:
Yup - but this is a common practice and it's bad ux to have developers write this every time - is it something we should add in the framework as a standard practice? cc @andrei-marinica